### DISCUSSION TITLE
Does SumatraPDF execute any kind of scripts?

### DISCUSSION THREAD

**Discussion by rigoxyz** [sumatrapdfreader/sumatrapdf#3811] [General]
Does SumatraPDF execute any kind of scripts from opened PDFs? If yes, how to disable it? Thank you :)

**Comment by GitHubRulesOK** [comment]
Generally NO that's always been the case of some users contention
it cannot run dynamic forms or other scripted behaviours EXCEPT if it is linked to external source / application So for example 
- it will use GhostScript if installed for PostScript running
- if you link on a link via embeded phishing a dark URL you will be transported by that to the dark side so dont click what you dont trust.

**Comment by kjk** [comment]
No, Sumatra doesn't execute any scripts.

**Reply by GitHubRulesOK** [reply]
I was going to add
- chm is a hybrid and thus exploitable, so really for safety it should by default not use MS render.  
 i.e. be set to `UseFixedPageUI = true`

for increased safety restrictions.ini may help

**Reply by rigoxyz** [reply]
Where can I find full list of options for sumatrapdfrestrict.ini?

**Reply by GitHubRulesOK** [reply]
@rigoxyz in the manual !
https://www.sumatrapdfreader.org/docs/Configure-for-restricted-use
